frame/referenda/src/types.rs

```diff
@@ -460,6 +485,21 @@ impl Curve {
 					.and_then(|term| (term - *x_offset).try_into_perthing().ok())
 					.unwrap_or_else(Perbill::one)
 			},
+			Self::ExponentialDecay { n_0, half_time, offset } => {
```

Comment:
I think this can be beautified to:

```rust

let ln_2 = ln(I64F64::from_num(2i32).expect("ln(2) is valid; qed");

offset
	.checked_sub(y.into())
	.and_then(|offset_minus_y| n_0.checked_div(offset_minus_y))
	.and_then(|inner_log| ln(-inner_log).ok())
	.and_then(|log_term| half_time.checked_mul(log_term))
	.and_then(|numerator| numerator.checked_div(ln_2))
	.unwrap_or(I64F64::from_num(1))
	.into()
```

Note: as `ln_2` is a closure, it will not be evaluated, if it is not actually called and will therefore not incur any runtime overhead.